helix-view/src/clipboard.rs

```diff
@@ -37,18 +69,24 @@ pub fn get_clipboard_provider() -> Box<dyn ClipboardProvider> {
         command_provider! {
             paste => "wl-paste", "--no-newline";
             copy => "wl-copy", "--type", "text/plain";
+            primary_paste => "wl-paste", "-p", "--no-newline";
```

Comment:
Given that the config says `middle_click_paste`, I'd rename all references in this file from `primary` to `middle_click` too.

Reply:
In terms of window systems (both X11 and Wayland) it's named `primary selection`, as well as in tools used for access to it. Config option is for UI actually, and we still have commands for accessing both clipboards, so setting actually disables mouse behavior (copy on select and paste on middle click), not the buffer.